### ISSUE TITLE
dual supervision for sparse/dense images

### ISSUE THREAD

**Issue by SherlockHolmes221** [cxliu0/PET#12]
why need dual supervision for sparse/dense images

In loss fuctuin dual supervision for sparse/dense images
loss_ce_sp = (raw_ce_loss * weights * div_mask)[sp_idx].sum() / ((weights * div_mask)[sp_idx].sum() + eps)
loss_ce_ds = (raw_ce_loss * weights * div_mask)[ds_idx].sum() / ((weights * div_mask)[ds_idx].sum() + eps)

thanks

**Comment by cxliu0** [comment]
As explained in the paper (Page 6), we use dual supervision to prevent the loss from being diluted by samples with few people, because such samples yield small training losses.

**Comment by SherlockHolmes221** [comment]
thanks
